packages/@aws-cdk/aws-appmesh/lib/http-route-path-match.ts

```diff
@@ -95,3 +95,153 @@ class HttpRouteWholePathMatch extends HttpRoutePathMatch {
+   *   and you want the route to match requests to "my-service.local/metrics", your prefix should be "/metrics".
+   * @param rewriteTo Specify either disabling automatic rewrite or rewriting to specified prefix path.
+   *   To disable automatic rewrite, provide `''`.
+   *   As a default, request's URL path is automatically rewritten to '/'.
+   */
+  public static startsWith(prefixPathMatch: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRoutePrefixPathMatch(prefixPathMatch, rewriteTo);
+  }
+
+  /**
+   * The value of the path must match the specified value exactly.
+   * The provided `path` must start with the '/' character.
+   *
+   * @param path the exact path to match on
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   *   As a default, retains original request's URL path.
+   */
+  public static exactly(path: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ exact: path }, rewriteTo);
+  }
+
+  /**
+   * The value of the path must match the specified regex.
+   *
+   * @param regex the regex used to match the path
+   * @param rewriteTo the value to substitute for the matched part of the path of the gateway request URL
+   *   As a default, retains original request's URL path.
+   */
+  public static regex(regex: string, rewriteTo?: string): HttpGatewayRoutePathMatch {
+    return new HttpGatewayRouteWholePathMatch({ regex }, rewriteTo);
+  }
+
+  /**
+   * Returns the gateway route path match configuration.
+   */
+  public abstract bind(scope: Construct): HttpGatewayRoutePathMatchConfig;
+}
+
+class HttpGatewayRoutePrefixPathMatch extends HttpGatewayRoutePathMatch {
+  constructor(
+    private readonly prefixPathMatch: string,
+    private readonly rewriteTo?: string,
+  ) {
+    super();
+
+    if (prefixPathMatch[0] !== '/') {
+      throw new Error('Prefix path for the match must start with \'/\', '
+        + `got: ${prefixPathMatch}`);
+    }
+
+    if (rewriteTo && rewriteTo !== '') {
+      if (prefixPathMatch[prefixPathMatch.length - 1] !== '/') {
+        throw new Error('When prefix path for the rewrite is specified, prefix path for the match must end with \'/\', '
+          + `got: ${prefixPathMatch}`);
+      }
+      if (rewriteTo[0] !== '/' || rewriteTo[rewriteTo.length - 1] !== '/') {
+        throw new Error('Prefix path for the rewrite must start and end with \'/\', '
+          + `got: ${rewriteTo}`);
+      }
+    }
+  }
+
+  bind(_scope: Construct): HttpGatewayRoutePathMatchConfig {
+    return {
+      prefixPathMatch: this.prefixPathMatch,
+      prefixPathRewrite: this.rewriteTo === '' || this.rewriteTo
+        ? {
+          defaultPrefix: this.rewriteTo === '' ? 'DISABLED' : undefined,
+          value: this.rewriteTo === '' ? undefined : this.rewriteTo,
+        }
+        : undefined,
```

Comment:
I believe you can simplify this:

```suggestion
      prefixPathRewrite: this.rewriteTo === undefined
        ? : undefined
        : {
          defaultPrefix: this.rewriteTo === '' ? 'DISABLED' : undefined,
          value: this.rewriteTo === '' ? undefined : this.rewriteTo,
        },
```

Reply:
It totally did!